test/UtilTests/DealerTest.cpp

```diff
@@ -0,0 +1,101 @@
+#include <gtest/gtest.h>
+#include <include/roboteam_ai/utilities/Dealer.h>
+#include <include/roboteam_ai/world_new/World.hpp>
+#include "../helpers/WorldHelper.h"
+#include "../helpers/FieldHelper.h"
+
+using namespace rtt::world_new;
+using namespace rtt::world_new::view;
+using namespace rtt::world_new::robot;
+
+namespace rtt::ai {
+
+/*
+ * @brief Fixture for the dealing test
+ *
+ * Every Test case will be booted with a world with 3+1 robots and no ball.
+ */
+class DealerTest : public ::testing::Test {
+ protected:
+  // DO NOT OVERRIDE STYLING ACCORDING TO CLANG TIDY
+  virtual void SetUp() {
+      int yellow = 1, blue = 3;
+      if (World::instance()->getSettings().isYellow()) {
+          yellow = 3;
+          blue = 1;
+      }
+      auto protoWorld =
+          testhelpers::WorldHelper::getWorldMsg(yellow, blue, false, testhelpers::FieldHelper::generateField());
+      World::instance()->updateWorld(protoWorld);
+  }
+};
+
+/*
+ * @brief Mock class for Dealer.
+ *
+ * The getDefaultFlagScores() function can be overridden,
+ * Since it brings in a lot of uncertainty from world and field.
+ */
+class MockDealer : public Dealer {
+ public:
+  explicit MockDealer(v::WorldDataView world) : Dealer(world, nullptr) {}
+  double getDefaultFlagScores(const RobotView &robot, const Dealer::DealerFlag &flag) override {
+      return robot->getId() + 1;
+  }
+  // MOCK_METHOD(double, getDefaultFlagScores, (const RobotView &robot, const Dealer::DealerFlag &flag), (override));
```

Comment:
?

Reply:
yeah that was the old Google mock method